src/DynamicExpresso.Core/Parsing/Parser.cs

```diff
@@ -1228,7 +1228,7 @@ private Expression ParseMethodGroupInvocation(MethodGroupExpression methodGroup,
 				applicableMethods = FindBestMethod(candidates.Select(_ => _.Method), args);
 
 			if (applicableMethods.Length == 0)
```

Comment:
Given the issue description I thought that we should throw `AmbiguousDelegateInvocation` when  `applicableMethods.Length > 1` . Why we throw when `== 0`?
Maybe I missing something .

@metoule  maybe you can help us ;-)

Reply:
That's precisely the issue I highlighted, and why it's actually quite difficult to fix properly :D 
The issue resides in the `FindBestMethod` method. When there are multiple applicable overloads, they're filtered thanks to the `MethodHasPriority` method:

```c#
private static MethodData[] FindBestMethod(IEnumerable<MethodData> methods, Expression[] args)
{
  // empty array if there are no applicable methods
  var applicable = methods
    .Where(m => CheckIfMethodIsApplicableAndPrepareIt(m, args))
    .ToArray();
  if (applicable.Length > 1)
  {
    // also returns an empty array if no overload has priority (ie. if there are multiple applicable methods)!
    return applicable
      .Where(m => applicable.All(n => m == n || MethodHasPriority(args, m, n)))
      .ToArray();
  }

  return applicable;
}
```

The issue is that the caller of `FindBestMethod` doesn't know if the resulting array is empty because no function was found, or if multiple applicable overloads were found.

We might be able to fix the issue like this, but it would require checking all calls to `FindBestMethod` to be sure:

```c#
private static MethodData[] FindBestMethod(IEnumerable<MethodData> methods, Expression[] args)
{
  // empty array if there are no applicable methods
  var applicable = methods
    .Where(m => CheckIfMethodIsApplicableAndPrepareIt(m, args))
    .ToArray();
  if (applicable.Length > 1)
  {
    var bestCandidates = applicable
      .Where(m => applicable.All(n => m == n || MethodHasPriority(args, m, n)))
      .ToArray();
    if (bestCandidates.length > 0)
      return bestCandidates; // should always have a length of 1 
  }

  return applicable;
}
```

Reply:
I like it. Thank you for the explanation. @sagilio  Do you think you can review the code to follow this suggestion?

Reply:
Thank you for the explanation, I already understand why to return 0 length array here. I think #231 has fixed this well, and will not have much performance impact when the overloads count is not very large.